ui/app/routes/vault/cluster/secrets/backend/secret-edit.js

```diff
@@ -35,15 +38,27 @@ export default Route.extend(UnloadModelRoute, {
     // perhaps in the future we could recurse _for_ users, but for now, just kick them
     // back to the list
     const { secret } = this.paramsFor(this.routeName);
-    const parentKey = utils.parentKeyForKey(secret);
-    const mode = this.routeName.split('.').pop();
-    if (mode === 'edit' && utils.keyIsFolder(secret)) {
-      if (parentKey) {
-        return this.transitionTo('vault.cluster.secrets.backend.list', parentKey);
-      } else {
-        return this.transitionTo('vault.cluster.secrets.backend.list-root');
+    return this.buildModel(secret).then(() => {
```

Comment:
Here should we redirect before doing the OpenAPI hydration?

Reply:
Does the other route need the hydration? I feel like if it should change routes and it doesn't then we should just transition